electron/app/jobs/save-to-db/index.ts

```diff
@@ -317,53 +326,65 @@ export class SaveToDbJob {
+                                            ? { clientProjectId: summary.id }
+                                            : {}),
+                                        ...(summary.type === 'client'
+                                            ? { clientId: summary.id }
+                                            : {}),
+                                        technologyId: null, // TODO: get technology by extracting the file extension in the window title of VS Code events
+                                        workType: summary.type,
+                                        userId: user.id,
+                                        status: 'confirmed',
+                                        approvalStatus: 'auto',
+                                        billableToClient: false,
+                                        source: 'automatic from GitStart DevTime app',
+                                    })),
+                                },
+                            });
+
+                            if ((insert.errors?.length ?? 0) > 0) {
+                                // ignore the error if we've already added worklogs at those times
+                                if (
+                                    !insert.errors.find((error) =>
+                                        error.message.includes('UQ_USER_WORK_LOG_NON_OVERLAPPING'),
+                                    )
+                                ) {
+                                    throw new Error(
+                                        `Error inserting user worklog | ${JSON.stringify(
+                                            insert.errors,
+                                        )}`,
+                                    );
+                                }
                             }
-                        }
 
-                        const returningLength = update.data.insert_user_work_logs.returning.length;
-                        this.lastSavedSummary = {
-                            ...grouppedSummary[grouppedSummary.length - 1],
-                            worklogId:
-                                update.data.insert_user_work_logs.returning[returningLength - 1].id,
-                        };
+                            const returningLength =
+                                insert.data.insert_user_work_logs.returning.length;
+                            this.lastSavedSummary = {
+                                ...grouppedSummary[grouppedSummary.length - 1],
+                                worklogId:
+                                    insert.data.insert_user_work_logs.returning[returningLength - 1]
+                                        .id,
+                            };
+                        }
+                    } catch (err) {
+                        if (!err.message.includes('UQ_USER_WORK_LOG_NON_OVERLAPPING')) {
+                            throw err;
+                        }
```

Comment:
if the error message doesn't contain UQ_USER_WORK_LOG_NON_OVERLAPPING, then its a real error and we want the code to fail, otherwise, we continue onto updating the events with is `isSummarized = true`